### ISSUE TITLE
CLDepthConcatenateLayer bad behavior

### ISSUE THREAD

**Issue by shaharf1** [ARM-software/ComputeLibrary#365]
We encounter a bad behavior related to _CLDepthConcatenateLayer_ on version 18.01.
It is seen with the code below.

We built a simple network , which concatenates ‘_data_tensor1’_ and ‘_data_tensor2’_ , into ‘_concat_tensor’_, 
and then uses ‘_CLActivationLayer’_ to multiply all values by -1. 
The final output is in ‘_neg_tensor’_.

The data in _‘concat_tensor’_ (after concatenation) is as expected.
However,  in ‘_neg_tensor’_ (after the -1 multiplication) only the first half of the data is good, and the other half is all zeros.

However, if we write the data directly to ‘_concat_tensor’_ , without using ‘_CLDepthConcatenateLayer’_, then the output in ‘_neg_tensor’_ is OK.
Is it a bug?

```
CLActivationLayer * neg_p;
CLDepthConcatenateLayer  * concat_p;

CLTensor data_tensor1;
CLTensor data_tensor2;
CLTensor concat_tensor;
CLTensor neg_tensor;


const TensorShape data_tensor_shape((unsigned int)240, (unsigned int)416, (unsigned int)3);
data_tensor1.allocator()->init(TensorInfo(data_tensor_shape, 1, DataType::F32));
data_tensor2.allocator()->init(TensorInfo(data_tensor_shape, 1, DataType::F32));

const TensorShape neg_tensor_shape((unsigned int)240, (unsigned int)416, (unsigned int)6);
concat_tensor.allocator()->init(TensorInfo(neg_tensor_shape, 1, DataType::F32));
neg_tensor.allocator()->init(TensorInfo(neg_tensor_shape, 1, DataType::F32));


data_tensor1.allocator()->allocate();
data_tensor2.allocator()->allocate();
concat_tensor.allocator()->allocate();
neg_tensor.allocator()->allocate();


concat_p = new CLDepthConcatenateLayer();

std::vector <ICLTensor * > concat_vec(2);
concat_vec[0] = &data_tensor1;
concat_vec[1] = &data_tensor2;
concat_p = new CLDepthConcatenateLayer();
concat_p->configure(concat_vec, &concat_tensor);

neg_p = new CLActivationLayer();
neg_p->configure(&concat_tensor, &neg_tensor, ActivationLayerInfo(ActivationLayerInfo::ActivationFunction::LINEAR  , -1 , 0));


copyDataToTensor((uint8_t *)input_image , &data_tensor1);
copyDataToTensor((uint8_t *)input_image , &data_tensor2);
concat_p->run();
neg_p->run();
```

Thanks.

**Comment by AnthonyARM** [comment]
Hi,

The sequence is wrong : you need to do
Init
Configure 
Allocate 
Run 

If you do allocate before configure then your buffers will be allocated without padding and therefore the library will only be able to process part of the tensors (leading to incomplete / invalid output tensors)

**Comment by shaharf1** [comment]
Are you sure?
I changed the sequence and I still get the same result.

Thanks.

**Comment by GeorgeARM** [comment]
Hello @shaharf1,
It seems this to be an issue from our side.
We will address it asap.
In the meantime you can use subtensors to perform the same functionality and have a `zero-copy` behavior.

**Comment by GeorgeARM** [comment]
Closing. Should have been resolved in 18.03. Reopen if needed.
